src/lib/Tests/RepositoryForms/Form/Processor/PreviewFormProcessorTest.php

```diff
@@ -0,0 +1,307 @@
+    }
+
+    public function testProcessPreview()
+    {
+        $languageCode = 'cyb-CY';
+        $contentDraftId = 123;
+        $url = 'http://url';
+        $fieldDefinitionIdentifier = 'identifier_1';
+        $fieldDataValue = 'some_value';
+
+        /**
+         * $data variable in PreviewFormProcessor classs.
+         */
+        $contentStruct = $this->generateContentStruct($languageCode, $fieldDefinitionIdentifier, $fieldDataValue);
+
+        $contentDraft = $this->generateContentDraft($contentDraftId, $languageCode);
+
+        $contentService = $this->generateContentServiceMock($contentStruct, $contentDraft);
+
+        $urlGenerator = $this->generateUrlGeneratorMock($contentDraft, $languageCode, $url);
+
+        $previewFormProcessor = $this->createPreviewFormProcessor($contentService, $urlGenerator);
+
+        $config = $this->generateConfigMock($languageCode);
+
+        $form = $this->generateFormMock($config);
+
+        $event = new FormActionEvent($form, $contentStruct, 'fooAction');
+
+        $previewFormProcessor->processPreview($event);
+
+        $this->assertEquals(new RedirectResponse($url), $event->getResponse());
+    }
+
+    public function testProcessPreviewHandleExceptionWithNew()
+    {
+        $languageCode = 'cyb-CY';
+        $contentDraftId = 123;
+        $url = 'http://url';
+        $fieldDefinitionIdentifier = 'identifier_1';
+        $fieldDataValue = 'some_value';
+
+        $contentStruct = $this->generateContentStruct($languageCode, $fieldDefinitionIdentifier, $fieldDataValue);
+
+        $config = $this->generateConfigMock($languageCode);
+
+        $form = $this->generateFormMock($config);
+
+        $event = new FormActionEvent($form, $contentStruct, 'fooAction');
+
+        $contentDraft = $this->generateContentDraft($contentDraftId, $languageCode);
+        $contentService = $this->createMock(ContentService::class);
+        $contentService
+            ->expects(self::once())
+            ->method('createContent')
+            ->will($this->throwException(new class('Location not found') extends \Exception {
+            }));
+
+        $urlGenerator = $this->generateUrlGeneratorForContentEditUrlMock($contentDraft, $languageCode, $url);
+        $notificationHandler = $this->generateNotificationHandlerMock();
+        $translator = $this->generateTranslatorMock();
+        $previewFormProcessor = $this->createPreviewFormProcessor($contentService, $urlGenerator, $notificationHandler, $translator);
+
+        $previewFormProcessor->processPreview($event);
+
+        $this->assertEquals(new RedirectResponse($url), $event->getResponse());
+    }
+
+    public function testSubscribedEvents()
+    {
+        $previewFormProcessor = $this->createPreviewFormProcessor();
+
+        $this->assertSame([ContentEditEvents::CONTENT_PREVIEW => ['processPreview', 10]], $previewFormProcessor::getSubscribedEvents());
+    }
+
+    protected function createUntouchableMock(string $class): MockObject
```

Comment:
What's the thinking behind untouchable mock? Any particular reason why you are mocking a class which should never be used by implementation?

Reply:
Yes, if we have situation like in `ProcessPreview` method, where we want to be sure, that in "positive path" we won't call any method from `notificationHandler`or `translator`, this is usecase for "UntouchableMock". E.g. if someone accidentally delete try-catch block.

Reply:
Sorry but you are restricting implementation details in the test cases which should NEVER be the case. Right now you are making any refactoring really hard because small change can make tests fail. How not using `$translator` is a valid assertion from the test standpoint?